File: day1/python/solution-part-two.py
with open('../input.txt', 'r') as handle:
    text = handle.read().splitlines()
    i = 0
    length = len(text)
    a = []
    final = []
    while i < length:
      line = text[i]
      if not line:
        final.append(sum(a))
        a = []
      else:
        a.append(int(line))
      i += 1
    print(sum(sorted(final, reverse=True)[:3]))

File: day1/ruby/solution-part-two.rb
file = File.open('input.txt')
file_data = file.readlines.map(&:chomp)
a = []
final = []
(0..file_data.length).each do |i|
  line = file_data[i]
  if line.nil? || line.empty?
    final.push(a.sum)
    a = []
  else
    a.push(line.to_i)
  end
end

puts final.sort {|x, y| y <=> x}.first(3).sum
